File: backend/src/main/java/com/technicalescaperoom/backend/controller/publicapi/PublicLeaderboardController.java
package com.technicalescaperoom.backend.controller.publicapi;

import com.technicalescaperoom.backend.dto.publicapi.PublicLeaderboardDto;
import com.technicalescaperoom.backend.service.admin.LeaderboardService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/public/events/{eventId}")
@RequiredArgsConstructor
public class PublicLeaderboardController {

    private final LeaderboardService leaderboardService;

    @GetMapping("/leaderboard")
    public ResponseEntity<PublicLeaderboardDto> getPublicLeaderboard(@PathVariable Long eventId) {
        PublicLeaderboardDto dto = leaderboardService.getPublicLeaderboard(eventId);
        return ResponseEntity.ok(dto);
    }
}

File: backend/controllers/publicLeaderboardController.js
const leaderboardService = require('../services/admin/leaderboardService');

class PublicLeaderboardController {
  async getPublicLeaderboard(req, res, next) {
    try {
      const eventId = parseInt(req.params.eventId, 10);
      const dto = await leaderboardService.getPublicLeaderboard(eventId);
      res.json(dto);
    } catch (err) {
      next(err);
    }
  }
}

module.exports = new PublicLeaderboardController();
